code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/service/impl/RancherConfigToComposeFormatter.java

```diff
@@ -0,0 +1,7 @@
+package io.cattle.platform.servicediscovery.service.impl;
+
+import io.cattle.platform.servicediscovery.resource.ServiceDiscoveryConfigItem;
+
+public interface RancherConfigToComposeFormatter {
+    public Object format(ServiceDiscoveryConfigItem item, Object valueToTransform);
```

Comment:
Can we add a comment or javadoc to mention that it returns null if no special formatting is necessary?  I actually think moving forward, we should probably do that for almost all interfaces.  But I'm sure people will want to debate it.

For this function, my initial thought was it would just return valueToTransform itself if no special formatting was done.  So I was confused when it returned null for almost all items.  Until I saw how it was being used in the caller.